File: xml/AppDelivery/app/src/main/java/com/brenoedl/appdelivery/FormLogin.java
package com.brenoedl.appdelivery;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.FirebaseNetworkException;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseAuthInvalidCredentialsException;
import com.google.firebase.auth.FirebaseAuthWeakPasswordException;
import com.google.firebase.auth.FirebaseUser;

import java.util.Objects;

public class FormLogin extends AppCompatActivity {
    private EditText etEmail, etSenha;
    private Button btEntrar;
    private TextView tvMensagemErro, tvCriarConta;
    private ProgressBar pbEntrar;
    private InputMethodManager imm;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_form_login);
        Objects.requireNonNull(getSupportActionBar()).hide();

        iniciarComponentes();

        btEntrar.setOnClickListener(view -> {
            String email = etEmail.getText().toString().trim();
            String senha = etSenha.getText().toString();

            if (email.isEmpty() || senha.isEmpty()) {
                tvMensagemErro.setText(getString(R.string.erro_campo));;
            } else {
                tvMensagemErro.setText("");
                imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
                autenticarUsuario(email, senha);
            }
        });

        tvCriarConta.setOnClickListener(view -> {
            Intent intent = new Intent(FormLogin.this, FormCadastro.class);
            startActivity(intent);
        });
    }

    @Override
    protected void onStart() {
        super.onStart();
        FirebaseUser usuarioAtual = FirebaseAuth.getInstance().getCurrentUser();
        if (usuarioAtual != null) {
            iniciarTelaProdutos();
        };
    }

    private void autenticarUsuario(String email, String senha) {
        FirebaseAuth.getInstance().signInWithEmailAndPassword(email, senha).addOnCompleteListener(task -> {
            if (task.isSuccessful()) {
                pbEntrar.setVisibility(View.VISIBLE);
                new Handler(Looper.getMainLooper()).postDelayed((this::iniciarTelaProdutos), 3000);
            } else {
                String erro;
                try {
                    throw Objects.requireNonNull(task.getException());
                } catch (FirebaseAuthWeakPasswordException e) {
                    erro = getString(R.string.erro_senha_pequena);
                } catch (FirebaseAuthInvalidCredentialsException e) {
                    erro = getString(R.string.erro_email);
                } catch (FirebaseNetworkException e) {
                    erro = getString(R.string.erro_internet);
                } catch (Exception e) {
                    erro = getString(R.string.erro_login);
                }
                tvMensagemErro.setText(erro);
            }
        });
    }

    private void iniciarTelaProdutos() {
        Intent intent = new Intent(FormLogin.this, ListaProdutos.class);
        startActivity(intent);
        finish();
    }

    private void iniciarComponentes() {
        etEmail = findViewById(R.id.etEmail);
        etSenha = findViewById(R.id.etSenha);
        tvMensagemErro = findViewById(R.id.tvMensagemErro);
        btEntrar = findViewById(R.id.btEntrar);
        tvCriarConta = findViewById(R.id.tvCriarConta);
        pbEntrar = findViewById(R.id.pbEntrar);
        imm = (InputMethodManager) getSystemService(INPUT_METHOD_SERVICE);
    }
}

File: xml/Kotlin/LojaVirtualCliente/app/src/main/java/com/brenoedl/lojavirtualcliente/activitys/formlogin/FormLogin.kt
package com.brenoedl.lojavirtualcliente.activitys.formlogin

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.os.Handler
import android.view.View
import android.view.inputmethod.InputMethodManager
import androidx.appcompat.app.AppCompatActivity
import com.brenoedl.lojavirtualcliente.R
import com.brenoedl.lojavirtualcliente.activitys.formcadastro.FormCadastro
import com.brenoedl.lojavirtualcliente.activitys.telaprincipalprodutos.TelaPrincipalProdutos
import com.brenoedl.lojavirtualcliente.databinding.ActivityFormLoginBinding
import com.google.android.material.snackbar.Snackbar
import com.google.firebase.FirebaseNetworkException
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseAuthInvalidCredentialsException
import com.google.firebase.auth.FirebaseAuthInvalidUserException
import com.google.firebase.auth.FirebaseAuthWeakPasswordException
import com.brenoedl.lojavirtualcliente.dialog.DialogCarregando
import com.google.android.gms.common.util.Hex
import com.mercadopago.android.px.internal.features.pay_button.PayButton

class FormLogin : AppCompatActivity() {
    private lateinit var binding: ActivityFormLoginBinding
    private val dialogCarregando = DialogCarregando(this)
    private lateinit var imm: InputMethodManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityFormLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)
        imm = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager

        binding.etEmail.requestFocus()

        binding.tvCadastro.setOnClickListener {
            val intent = Intent(this, FormCadastro::class.java)
            startActivity(intent)
        }

        binding.bEntrar.setOnClickListener {
            val email = binding.etEmail.text.toString().trim()
            val senha = binding.etSenha.text.toString()

            if (email.isEmpty() || senha.isEmpty()) {
                val snackbar = Snackbar.make(it,getString(R.string.erro_campo), Snackbar.LENGTH_SHORT)
                snackbar.setBackgroundTint(Color.RED)
                snackbar.show()
            } else {
                logarUsuario(email, senha, it)
            }
        }
    }

    override fun onStart() {
        super.onStart()

        val usuarioAtual = FirebaseAuth.getInstance().currentUser
        if (usuarioAtual != null) {
            navegarTelaPrincipal()
        }
    }

    private fun navegarTelaPrincipal() {
        val intent = Intent(this, TelaPrincipalProdutos::class.java)
        startActivity(intent)
        finish()
    }

    private fun logarUsuario(email: String, senha: String, view: View) {
        FirebaseAuth.getInstance().signInWithEmailAndPassword(email, senha)
            .addOnCompleteListener {
                if (it.isSuccessful) {
                    imm.hideSoftInputFromWindow(view.windowToken, 0)

                    val snackbar = Snackbar.make(binding.root, getString(R.string.sucesso_login), Snackbar.LENGTH_SHORT)
                    snackbar.setBackgroundTint(Color.BLUE)
                    snackbar.show()
                    dialogCarregando.iniciarCarregamentoAlertDialog()

                    Handler(mainLooper).postDelayed({
                        navegarTelaPrincipal()
                        dialogCarregando.liberarAlertDialog()
                    }, 3000)
                }
            }.addOnFailureListener { exception ->
                val erro = when(exception) {
                    is FirebaseAuthWeakPasswordException -> getString(R.string.erro_senha)
                    is FirebaseAuthInvalidUserException -> getString(R.string.erro_usuario)
                    is FirebaseAuthInvalidCredentialsException -> getString(R.string.erro_email)
                    is FirebaseNetworkException -> getString(R.string.erro_internet)
                    else -> getString(R.string.erro_login)
                }
                val snackbar = Snackbar.make(binding.root, erro, Snackbar.LENGTH_SHORT)
                snackbar.setBackgroundTint(Color.RED)
                snackbar.show()
            }
    }
}